src/here-studio.ts

```diff
@@ -0,0 +1,342 @@
+    for (let i=0; i < clonedProjectData.layers.length; i++) {
+
+        //Get the current layer
+        let currentLayer = clonedProjectData.layers[i];
+
+        //Check if the layer has tags eg. Building Footprints tags -> In Such cases copy the whole currentLayer as is -> Otherwise copy the space from the user
+        if (currentLayer.meta
+            && currentLayer.meta.tags
+            && currentLayer.meta.tags.length > 0) {
+
+            //Update the project layer as is without any modifications
+            updatedLayersData.push(currentLayer)
+        }
+        else {
+
+            //Download the space locally and reference the space for current user
+            let geoSpaceIDToCopy = currentLayer.geospace.id;
+            console.log(`\nCopying layer [${i+1}] : ${geoSpaceIDToCopy} from published project`)
+
+            //Get the space title and description from the base space
+            const url = `/hub/spaces/${geoSpaceIDToCopy}?clientId=cli`
+            const response = await execute(url,"GET", "application/json", "", publishersToken);
+
+            //Copy the contents of the space config title and description for currentUser's XYZ spaces
+            let spaceConfigOptions = {
+                title : response.body.title,
+                message : response.body.description
+            }
+
+            //Get the original count of features to download from statistics API - getStatisticsData
+            let spaceStatsData = await getStatisticsData(geoSpaceIDToCopy, publishersToken);
+
+            //Download the GeospaceID from published project with the publisher's token -  Download the space from GET Search and save it in local temp file using /iterate features API
+            let geoSpaceDownloadOptions = {
+                token : publishersToken,
+                limit : spaceStatsData.count.value // Fetch all features from base spaceID
+            }
+            let geoSpaceData = await getSpaceDataFromXyz(geoSpaceIDToCopy, geoSpaceDownloadOptions);
+
+            //Check if GeoSpaceData is blank -> if yes move on to next one else create the file with that name
+            if ( spaceStatsData.count.value === 0) {
+                console.log("\nNo features are available to download");
+                //process.exit();
+            }
+            else {
+                let geoSpaceFileName = geoSpaceIDToCopy+".geojson";
+                await fs.writeFileSync(geoSpaceFileName, JSON.stringify(geoSpaceData));
+                console.log("Space '"+geoSpaceIDToCopy+"' downloaded locally")
+
+                //Create a new space for currentUser
+                let newSpaceData = await createSpace (spaceConfigOptions)
+                let currentGeoSpaceID = newSpaceData.id;
+
+                //Update the geospace id for current layer
+                currentLayer.geospace.id = currentGeoSpaceID;
+
+                let uploadOptions = {
+                    title: geoSpaceFileName,
```

Comment:
no need to give title and description here

Reply:
Updated - Have removed title and description from upload method as the title and description was given above using getSpaceMetaData method